cogs/fun.py

```diff
@@ -1,16 +1,272 @@
+            await ctx.send("Sorry, No Images Found.")
+
+    @command()
+    async def chuck(self, ctx: Context) -> t.Union[None, Message]:
+        """Get a random Chuck Norris joke"""
+        if randint(0, 1):
+            async with aiohttp.ClientSession() as session:
+                async with session.get("https://api.chucknorris.io/jokes/random") as r:
+                    joke = await r.json()
+                    return await ctx.send(joke["value"])
+        if ctx.guild:
+            if not ctx.channel.is_nsfw():
+                async with aiohttp.ClientSession() as session:
+                    async with session.get("http://api.icndb.com/jokes/random?exclude=[explicit]") as r:
+                        joke = await r.json()
+                        return await ctx.send(joke["value"]["joke"])
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get("http://api.icndb.com/jokes/random") as r:
+                joke = await r.json()
+                await ctx.send(joke["value"]["joke"].replace("&quote", '"'))
+
+    @command()
+    async def cat(self, ctx: Context) -> None:
+        """Random cat images. Awww, so cute! Powered by random.cat"""
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://aws.random.cat/meow", headers=self.user_agent) as r:
+                if r.status == 200:
+                    js = await r.json()
+                    em = discord.Embed(name="random.cat", colour=0x690E8)
+                    em.set_image(url=js["file"])
+                    await ctx.send(embed=em)
+                else:
+                    raise ServiceError(f"could not fetch cute cat :( (http {r.status})")
+
+    @command()
+    async def httpcat(self, ctx: Context, http_id: int) -> None:
+        """http.cat images - >>httpcat <http code>"""
+        if http_id in constants.http_codes:
+            httpcat_em = discord.Embed(name="http.cat", colour=0x690E8)
+            httpcat_em.set_image(url=f"https://http.cat/{http_id}.jpg")
+            await ctx.send(embed=httpcat_em)
+        else:
+            raise BadArgument("Specified HTTP code invalid")
+
+    @command()
+    async def fox(self, ctx: Context) -> None:
+        """Sends a random fox picture"""
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://randomfox.ca/floof/") as response:
+                picture = await response.json()
+                embed = discord.Embed(title="Fox", color=0x690E8)
+                embed.set_image(url=picture["image"])
+                await ctx.send(embed=embed)
+
+    @command()
+    async def dog(self, ctx: Context) -> None:
+        """Random doggos just because!"""
+
+        def decide_source() -> str:
+            n = random.random()
+            if n < 0.5:
+                return "https://random.dog/woof"
+            elif n > 0.5:
+                return "https://dog.ceo/api/breeds/image/random"
```

Comment:
No need to generate random float, random number of 1 or 0 is enough here.
```suggestion
            n = random.randint(0, 1)
            if n == 0:
                return "https://random.dog/woof"
            elif n == 1:
                return "https://dog.ceo/api/breeds/image/random"
```